beacon_node/beacon_chain/src/attestation_rewards.rs

```diff
@@ -218,15 +218,25 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
                         source_reward = *penalty;
                     }
                 }
+
+                let inactivity_score = state.get_inactivity_score(*validator_index)?;
+                let penalty_numerator = effective_balance.safe_mul(inactivity_score)?;
+                let penalty_denominator = spec
+                    .inactivity_score_bias
+                    .safe_mul(spec.inactivity_penalty_quotient_for_state(&state))?;
+                inactivity_penalty = penalty_numerator.safe_div(penalty_denominator)? as i64;
```

Comment:
line 119 below is important - you only assign an inactivity penalty if validator didn't vote target correctly:
https://github.com/sigp/lighthouse/blob/f2223feb21bb10dbdebce097c5cda384b1ff44a3/consensus/state_processing/src/per_epoch_processing/altair/rewards_and_penalties.rs#L119-L128

Perhaps you need to `inactivity_penalty` calculation to this `else if` block:
https://github.com/sigp/lighthouse/blob/4f756600d1e1569711f8623452435365bc5a037d/beacon_node/beacon_chain/src/attestation_rewards.rs#L215-L216

Other than this, it looks like you're on the right track!

Reply:
@jimmygchen I moved the calculation to the location suggested above.